src/util/locking/RedisResourceLocker.ts

```diff
@@ -110,4 +135,24 @@ export class RedisResourceLocker implements ResourceLocker {
   private getLockCount(): number {
     return this.lockList.size;
   }
+
+  /**
+   * This function is internally used to keep an acquired lock active, a wrapper class will handle expiration
+   * @param lock - the lock to be extended
+   */
+  private extendLockIndefinitely(lock: Lock): void {
+    setTimeout(async(): Promise<void> => {
+      if (this.lockList.get(lock.resource)) {
+        try {
+          const newLock = await this.redlock.extend(lock, 1000);
+          this.lockList.set(newLock.resource, newLock);
+          this.extendLockIndefinitely(newLock);
+          this.logger.debug(`Extended (Redis)lock for resource: ${newLock.resource}`);
+        } catch (error: unknown) {
+          // No error should be thrown cause this means the lock has simply been released
+          this.logger.debug(`Failed to extend this (Redis)lock for resource: ${lock.resource}, ${error}`);
+        }
+      }
+    }, 900);
```

Comment:
Better let this depend on the `ttl` constant, and maybe with a larger margin, e.g. `ttl/2`, since timeouts are a minimum, not a guaranteed interval.